toolkit/tools/pkg/imagecustomizerlib/imagehistory.go

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+)
+
+type ImageHistory struct {
+	BuildTime   string                    `yaml:"timestamp" json:"timestamp"`
+	ToolVersion string                    `yaml:"toolVersion" json:"toolVersion"`
+	ImageUuid   string                    `yaml:"imageUuid" json:"imageUuid"`
+	Config      imagecustomizerapi.Config `yaml:"config" json:"config"`
+}
+
+func addImageHistory(imageChroot *safechroot.Chroot, imageUuid string, baseConfigPath string, toolVersion string, buildTime string, config *imagecustomizerapi.Config) error {
+	var err error
+	logger.Log.Infof("Creating image customizer history file")
+
+	// Deep copy the config to avoid modifying the original config
+	configCopy, err := deepCopyConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to deep copy config while writing image history: %w", err)
+	}
+
+	err = modifyConfig(configCopy, baseConfigPath)
+	if err != nil {
+		return fmt.Errorf("failed to modify config while writing image history: %w", err)
+	}
+
+	customizerLoggingDirPath := filepath.Join(imageChroot.RootDir(), "/usr/share/image-customizer")
```

Comment:
It would be good to make `"/usr/share/image-customizer"` and `"history.json"` constants at the top of the file.